src/main/kotlin/com/sourcegraph/cody/chat/ui/LlmDropdown.kt

```diff
@@ -73,19 +69,14 @@ class LlmDropdown(
     val availableModels = models.map { it }.filterNot { it.model.isDeprecated() }
     availableModels.sortedBy { it.model.isCodyProOnly() }.forEach { addItem(it) }
 
-    val selectedFromChatState = chatModelFromState
-    val selectedFromHistory = HistoryService.getInstance(project).getDefaultLlm()
+    val defaultLlm =
+        DefaultLlmStorage.getInstance(project).get(CodyAccount.getActiveAccount()?.server)
 
     selectedItem =
-        availableModels.find {
-          it.model.id == model ||
-              it.model.id == selectedFromHistory?.model ||
-              it.model.id == selectedFromChatState?.id
-        } ?: models.firstOrNull()
-
-    // If the dropdown is already disabled, don't change it. It can happen
-    // in the case of the legacy commands (updateAfterFirstMessage happens before this call).
-    isEnabled = isEnabled && chatModelFromState == null
+        availableModels.find { it.model.id == model || it.model.id == defaultLlm?.id }
+            ?: models.firstOrNull()
+
+    isEnabled = true
```

Comment:
Previously we were disabled model selection after edit and not re-enabling it on retry.
Is that intentional change?